src/containers/billing/index.js

```diff
@@ -1,74 +1,270 @@
+import 'containers/billing/billing.scss';
+import { setBalance, setUsage } from 'redux/modules/transaction-group';
+
+const transactionRangeQuery =
+  gql`query usageTransactions($startDate: String!, $endDate: String!) {
+      credits(startDate: $startDate, endDate: $endDate) {
+        id,
+        paid_amount,
+        created,
+        type
+      }
+      debits(startDate: $startDate, endDate: $endDate) {
+        id,
+        amount,
+        created,
+        type
+      },
+    }`;
 
 const mapQueriesToProps = () => {
   return {
-    data: {
-      query: gql`query getTransactions($id: String!) {
-        user(id: $id) {
-          credits {
-            id,
-            amount,
-            created
+    paymentProcessor: {
+      query: gql`query {
+        paymentProcessor {
+          id,
+          name,
+          billingDate,
+          defaultPaymentMethod {
+            merchant,
+            lastFour,
+            id
           },
-          debits {
+          error
+        }
+      }`
+    },
+    transactions: {
+      query: gql`query {
+        credits {
+          id,
+          paid_amount,
+          created,
+          type
+        },
+        debits {
+          id,
+          amount,
+          created,
+          type
+        }
+      }`
+    }
+  }
+};
+
+const mapMutationsToProps = () => {
+  return {
+    removeCard: (paymentProcessorId, paymentMethodId) => {
+      return {
+        // TODO: maybe add `id,` to get query update?
+        mutation: gql`
+        mutation removePaymentMethod($paymentProcessorId: String!, $paymentMethodId: String!) {
+          removePaymentMethod(paymentProcessorId: $paymentProcessorId, paymentMethodId: $paymentMethodId) {
             id,
-            amount,
-            created
+            name,
+            billingDate,
+            defaultPaymentMethod {
+              merchant,
+              lastFour,
+              id
+            },
+            error
           }
+        }`,
+        variables: {
+          paymentProcessorId,
+          paymentMethodId
         }
-      }`,
-      variables: {
-        id: 'user1@example.com'
-      }
+      };
     }
   };
 };
 
+const mapDispatchToProps = {
+  setBalance,
+  setUsage
+};
+
+const mapStateToProps = ({transactionGroup: {balance, usage}}) => {
+  return {balance, usage};
+};
+
+let globalCounter = 0;
+
 @connect({
-  mapQueriesToProps
+  mapQueriesToProps,
+  mapMutationsToProps,
+  mapDispatchToProps,
+  mapStateToProps
 })
 
 export default class Billing extends Component {
-  getBalance() {
-    const {loading, user} = this.props.data;
+  componentWillReceiveProps(nextProps) {
+    globalCounter++;
+    if (globalCounter > 50) return;
+    console.log(globalCounter);
 
-    if (loading) {
-      return '';
+    const {balance, usage} = nextProps;
+    console.log('balance and usage: ', balance, usage);
+
+    if (!!balance && !!usage) {
+      return console.log('stopping');
```

Comment:
not quite - but good eye :D. This function is called every time the component receives new props... In this case, we are waiting for balance and usage data to come back from graphql... if it's not back yet then we want so skip the rest of the function because it requires this data. A return allows react to continue.